### PR TITLE
Updates to util functions and their tests

### PR DESCRIPTION
**Pull request by benhammondmusic**
## Description

- change the name of the state info merge function to reflect its ability to merge onto fips, names, or postal codes
- some linting / naming / type hinting
- adds _merge_pop function wrapper that allows for merging ACS 2019 only onto particular "current" years

## Motivation and Context

As part of #1851 I've made some adjustments to existing util functions and added some new ones. This PR extracts these changes into a separate PR to make changes easier to read/understand and ensure they're working properly before merging the rest of the data source specific code

## How has this been tested?

automated tests have been updated / added and are passing

## Types of changes

- New feature (non-breaking change which adds functionality or updates content)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by joshzarrabi — COMMENTED** [review]
This all looks good, before I approve would you mind running the `e2e_tests/scripts/ensure_datasets_equal.py` comparing the new covid dataset you created and the one on dev. Just to make sure there were no inadvertent changes.

File: python/ingestion/dataset_utils.py [hunk 1]
```diff
@@ -260,12 +266,46 @@ def generate_pct_relative_inequity_column(df, pct_share_col, pct_pop_col, pct_re
...
+    Parameters:
+        df: dataframe containing cols `time_period`, `state_fips`
+        pct_rel_inequity_col: string column name containing the previously calculated percents
+        rate_col: string column name for the rate column used to trigger the "nulling" filter
+    Returns:
+        df with relative inequity nulled for years that contain 0 rates for all groups
+    """
+
+    # detect non-zero 100ks in each grouping of PLACE/YEAR
+    df["year_state"] = df["time_period"] + df["state_fips"]
```
[the remark this answers is not in the document]
Comment (benhammondmusic) [reply python/ingestion/dataset_utils.py]: can't we lint to replace double with single quotes?  I always type double quotes by default for everything....

[the remark this answers is not in the document]
Comment (joshzarrabi) [reply python/ingestion/dataset_utils.py]: thats a good point, our python linter just checks for errors, doesnt actually change the code. IMO single quotes are better practice in python

**Comment by benhammondmusic** [comment]
> This all looks good, before I approve would you mind running the `e2e_tests/scripts/ensure_datasets_equal.py` comparing the new covid dataset you created and the one on dev. Just to make sure there were no inadvertent changes.

@joshzarrabi  looks like that's all it was; my BQ was building off of the old covid data
Screen Shot 2022-11-22 at 11 07 48 AM

**Review by joshzarrabi — APPROVED** [review]
Good work

**Pull request merged**